### DISCUSSION TITLE
Launch pipeline execution in Dagit programmatically

### DISCUSSION THREAD

**Discussion by ankxyz** [dagster-io/dagster#3048] [Q&A] [answered]
Can I simulate click on `Launch Execution` in **Dagit** UI programmatically? 

Moreover, it would be great to execute pipelines using Python API: run pipeline and wait until it finish.

Conclusion:
* execute pipeline so that **Dagit** will log pipeline run;
* wait until pipeline finish.

Thank you for advance.

**Comment by alangenfeld** [comment, marked as answer]
> Can I simulate click on Launch Execution in Dagit UI programmatically?

There is the `dagster pipeline launch` cli command. At this time there is not a convenient python API for launch.

> Moreover, it would be great to execute pipelines using Python API: run pipeline and wait until it finish.

https://docs.dagster.io/_apidocs/execution#dagster.execute_pipeline

To make sure it's reported to the instance you have set up, set it by doing `execute_pipeline(my_pipeline, instance=DagsterInstance.get())`. By default `execute_pipeline` uses an ephemeral instance under the assumption that most of its uses are in tests that shouldn't be logged to the instance.

**Reply by ankxyz** [reply]
Thank you very much! You helped me!
